bundles/framework/announcements/view/AnnouncementsHandler.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { StateHandler, controllerMixin, Messaging } from 'oskari-ui/util';
+import { Message } from 'oskari-ui';
+
+// Handler for announcements. Handles state and service calls.
+
+const getMessage = (key, args) => <Message messageKey={key} messageArgs={args} bundleKey='announcements' />;
+
+class ViewHandler extends StateHandler {
+    constructor () {
+        super();
+        this.state = {
+            panels: [],
+            modals: [],
+            updated: false,
+            announcements: [],
+            checked: false
+        };
+    }
+
+    getAnnouncements (callback) {
+        if (typeof callback !== 'function') {
+            return;
+        }
+
+        jQuery.ajax({
+            type: 'GET',
+            dataType: 'json',
+            url: Oskari.urls.getRoute('Announcements'),
+            success: function (pResp) {
+                callback(pResp);
+            },
+            error: function (jqXHR, textStatus) {
+                Messaging.error(getMessage('messages.getFailed'));
+            }
+        });
+    }
+
+    updatePanelsModals (panels, modals) {
+        this.updateState({
+            panels: panels,
+            modals: modals,
+            updated: true
+        });
+    }
+
+    setAnnouncementAsSeen (index, checked, id) {
+        if (checked) {
+            this.addToLocalStorageArray('oskari-announcements', id);
```

Comment:
Move the name of the localstorage variable to a constants file for announcements. Import it here and in the admin and use the name from that constant so the two bundles don't go out-of-sync if one is updated (making it more maintainable).